src/Tokenizer/Tokens.php

```diff
@@ -1085,18 +1175,26 @@ public function isMonolithicPhp()
             return false;
         }
 
-        // If code is not monolithic there is a great chance that first or last token is `T_INLINE_HTML`:
-        if ($this[0]->isGivenKind(T_INLINE_HTML) || $this[$size - 1]->isGivenKind(T_INLINE_HTML)) {
-            return false;
-        }
-
-        for ($index = 1; $index < $size; ++$index) {
-            if ($this[$index]->isGivenKind([T_INLINE_HTML, T_OPEN_TAG, T_OPEN_TAG_WITH_ECHO])) {
+        if (self::isLegacyMode()) {
+            // If code is not monolithic there is a great chance that first or last token is `T_INLINE_HTML`:
+            if ($this[0]->isGivenKind(T_INLINE_HTML) || $this[$size - 1]->isGivenKind(T_INLINE_HTML)) {
                 return false;
             }
+
+            for ($index = 1; $index < $size; ++$index) {
+                if ($this[$index]->isGivenKind([T_INLINE_HTML, T_OPEN_TAG, T_OPEN_TAG_WITH_ECHO])) {
+                    return false;
+                }
+            }
+
+            return true;
         }
 
-        return true;
+        if ($this->isTokenKindFound(T_INLINE_HTML)) {
+            return false;
+        }
+
+        return 1 >= ($this->countTokenKind(T_OPEN_TAG) + $this->countTokenKind(T_OPEN_TAG_WITH_ECHO));
```

Comment:
Yoda style in expressions using `<` or `>` is confusing IMO, I would swap the operands.